src/MDMScheduler.m

```diff
@@ -81,16 +81,33 @@ - (MDMSchedulerActivityState)activityState {
 }
 
 - (void)addPlan:(NSObject<MDMPlan> *)plan to:(id)target {
+  [self commonAddPlan:plan toTarget:target log:nil];
+}
+
+- (void)addPlan:(NSObject<MDMPlan> *)plan named:(NSString *)name to:(id)target {
+  NSParameterAssert(name.length > 0);
+  [self commonAddPlan:plan toTarget:target log:[[MDMTransactionLog alloc] initWithPlans:@[plan] target:target name:name removal:FALSE]];
```

Comment:
i find the `removal:FALSE` init param a little clunky. Especially since the `removal:TRUE` case is just omitted